weather.py

```diff
@@ -0,0 +1,440 @@
+                print(
+                    f"""        
+                        {i + 1} {max_pads[i]} {max_temp[i]}C
+                        {i + 1} {min_pads[i]} {min_temp[i]}C""")
+
+            else:
+                print(
+                    f"""
+                        {i + 1} {min_pads[i]}{max_pads[i]} {min_temp[i]}C - {max_temp[i]}C""")
+
+        self.report_num += 1
+
+    # Method to read lines from the file and return
+
+    def calculate_highest_helper(self, dic, high_temp, low_temp, high_humidity, high_date, low_date, humid_date):
+
+        if dic["Max TemperatureC"] != "":  # lst[1] contains high temp results in a line
+            if int(dic["Max TemperatureC"]) > high_temp:  # code to check highest
+                high_temp = int(dic["Max TemperatureC"])
+                high_date = dic["PKT"]
+
+        if dic["Min TemperatureC"] != "":  # lst[3] contains low temp results in a line
+            if int(dic["Min TemperatureC"]) < low_temp:  # code to check lowest
+                low_temp = int(dic["Min TemperatureC"])
+                low_date = dic["PKT"]
+
+        if dic["Max Humidity"] != "":
+            if int(dic["Max Humidity"]) > high_humidity:
+                high_humidity = int(dic["Max Humidity"])
+                humid_date = dic["PKT"]
+
+        return high_temp, high_date, low_temp, low_date, high_humidity, humid_date
+
+    def highest_per_month(self, year, month, high_temp, low_temp,
+                          high_humidity, high_date,
+                          low_date, humid_date):
+
+        file = self.files_path_prefix + year + "_" + month + ".txt"
+
+        if path.exists(file):
+            reader = CSVReader(file)
+
+            dictionary = reader.get_line()
+            while dictionary:
+                high_temp, high_date, low_temp, low_date, high_humidity, humid_date \
+                    = self.calculate_highest_helper(dictionary, high_temp, low_temp,
+                                                    high_humidity, high_date,
+                                                    low_date, humid_date)
+                dictionary = reader.get_line()
+
+            return True, high_temp, high_date, low_temp, low_date, high_humidity, humid_date
+        return False, high_temp, high_date, low_temp, low_date, high_humidity, humid_date
+
+    def calculate_highest(self, date):
+        high_temp = 0
+        high_humidity = 0
+        low_temp = sys.maxsize
+        high_date, low_date, humid_date = "", "", ""
+        exists = False
+
+        if len(date) < 2:  # To check if month is inputted or only year
+            for month in self.months:  # Running code for all 12 months and opening files that exist
+                if not exists:  # If no files found, keep checking
+                    exists, high_temp, high_date, low_temp, low_date, high_humidity,\
+                     humid_date = self.highest_per_month(date[0], month, high_temp,
```

Comment:
Instead of getting file existence information from `highest_per_month` check file availability before calling `highest_per_month` if a file doesn't exit skip the calculation for this month.

Reply:
done